File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> heap=new PriorityQueue<>();
        for(int num:nums){
            heap.add((long)num);
        }
        Long x,y;
        int count=0;
        while(heap.peek()<k){
            x=heap.poll();
            y=heap.poll();
            heap.add(Math.min(x,y)*2+Math.max(x,y));
            count++;
        }
        return count;
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        count=0
        heapq.heapify(nums)
        if len(nums)<2:
            return -1
        while nums[0]<k:
            x,y=heapq.heappop(nums),heapq.heappop(nums)
            heapq.heappush(nums,min(x,y)*2+max(x,y))
            count+=1
        return count
        